src/cppmangle.c

```diff
@@ -164,7 +164,8 @@ class CppMangleVisitor : public Visitor
                         else
                         {
                             dinteger_t val = e->toInteger();
-                            if (val < 0)
+                            // TODO: try and do this without knowing what dinteger_t type is
+                            if ((long)val < 0)
```

Comment:
> No, in fact it has to be otherwise the negation won't work.

This doesn't make sense to me BTW. The whole point of this code is to unset the sign bit. This should make the unsigned version the same as signed version.